internal/pkg/telemetry/span.go

```diff
@@ -1,19 +1,33 @@
 package telemetry
 
 import (
+	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
 )
 
-func EndSpan(span trace.Span, errPtr *error) {
+type Span interface {
+	End(errPtr *error, opts ...trace.SpanEndOption)
+	SetAttributes(kv ...attribute.KeyValue)
+}
+
+type span struct {
+	span trace.Span
+}
+
+func (s *span) SetAttributes(kv ...attribute.KeyValue) {
+	s.span.SetAttributes(kv...)
+}
+
+func (s *span) End(errPtr *error, opts ...trace.SpanEndOption) {
```

Comment:
V kode sme mali funkciu `telemetry.EndSpan`:
https://github.com/keboola/keboola-as-code/blob/944621848e70d409439237a21a86b86386610ece/internal/pkg/service/common/dependencies/public.go#L40-L44

Islo o to zapisat span co najkratsie, kedze ten OTEL interface je dost ukecany, vid nizsie `.RecordErrr(), .SetStatus(), .End`. Okrem toho sa nam stavalo, ze vzdy niekto zabudol trackovat chybu danej operacie (a skoro kazda moze skoncit chybou), preto mala funkcia `EndSpan` povinny argument `errPtr`.

-----------------

Nahradil som to metodou `span.End(errPtr)`